src/coin_test/data/datasaver.py

```diff
@@ -0,0 +1,57 @@
+"""Define the Datasaver class."""
+import os
+import pickle
+
+from .datasets import Dataset
+
+
+class Datasaver:
+    """Hold collections of datasets to save."""
+
+    def __init__(
+        self, name: str, datasets: list[Dataset] | list[list[Dataset]]
+    ) -> None:
+        """Initialize a Datasaver.
+
+        Args:
+            name (str): string name for datasaver
+            datasets (list[Dataset] | list[list[Dataset]]): collection of
+                datasets to save
+        """
+        self.name = "".join(x for x in name if x.isalnum() or x in ["_"])
+
+        if isinstance(datasets[0], Dataset):
+            self.datasets_lists = [[d] for d in datasets]
+        else:
+            self.datasets_lists = datasets
+
+    @staticmethod
+    def save(datasaver: "Datasaver", directory: str) -> str:
```

Comment:
I think it would be much more intuitive if these weren't static methods. Calling `datasaver.save(path)` nicer than `Datasaver.save(datasaver, path)`. Would testing be much more work?